pandas/_libs/lib.pyx

```diff
@@ -2031,10 +2033,15 @@ def maybe_convert_numeric(
         numeric array has no suitable numerical dtype to return (i.e. uint64,
         int32, uint8). If set to False, the original object array will be
         returned. Otherwise, a ValueError will be raised.
-
+    convert_to_nullable_integer : bool, default False
```

Comment:
can we just have a single argument 'convert_to_nullable' (e.g. if the results has float, int, or bool then its really easy to pick the appropriate type). I think this is way more complex with 2 args (e.g. many more paths)